python/lsst/ts/fbs/utils/maintel/make_fieldsurvey_scheduler.py

```diff
@@ -0,0 +1,128 @@
+# This file is part of ts_fbs_utils.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MakeFieldSurveyScheduler"]
+
+import typing
+
+from rubin_scheduler.scheduler.basis_functions import BaseBasisFunction
+from rubin_scheduler.scheduler.schedulers import CoreScheduler
+from rubin_scheduler.scheduler.surveys import BaseSurvey, FieldSurvey
+
+from ..data import field_survey_centers
+
+
+class MakeFieldSurveyScheduler:
+    """Construct Feature Based Scheduler configuration consisting of an
+    ensemble of field surveys, and optionally, other types of surveys.
+    """
+
+    def __init__(
+        self,
+        nside: int = 32,
+        ntiers: int = 1,
+    ) -> None:
+
+        self.nside = 32
+        self.surveys: typing.List[typing.List[BaseSurvey]] = [
+            [] for _ in range(0, ntiers)
+        ]
+
+    def _load_candidate_fields(self) -> typing.Dict:
+        """Load pointing center data for field surveys."""
+        name_fields, ra_fields, dec_fields = zip(*field_survey_centers.fields)
+        fields = {}
+        for name, ra, dec in zip(name_fields, ra_fields, dec_fields):
+            fields[name] = {"RA": ra, "dec": dec}
+        return fields
+
+    def add_field_surveys(
+        self,
+        tier: int,
+        json_block: str,
+        observation_reason: str,
+        science_program: str,
+        field_names: typing.List[str],
+        basis_functions: typing.List[BaseBasisFunction] = [],
+        **kwargs: typing.Any,
+    ) -> None:
+        """Add a list of field surveys to the scheduler configuration.
+
+        Parameters
+        ----------
+        tier: `int`
+            Tier index used to control prioritization of surveys.
+        json_block: `str`
+            JSON BLOCK used to perform the observations.
+        observation_reason: `str`
+            Purpose of the observation, e.g., "science"
+        science_program: `str`
```

Comment:
@tribeiro has suggested that json_block and science_program should be the same, and science_program is what should come from the FBS and then be used as the JSON BLOCK identifier .. as well as head into the ConsDB as "science_program".

Reply:
Agreed, let's pass the JSON BLOCK identified (BLOCK-320) at the ts_config_ocs level for this parameter.